<launch>
	<arg name="pkg_name" value="led_emitter"/>
	<arg name="src_name" value="led_emitter_node.py"/>
	<arg name="node_name" value="led_emitter_node"/>
	<arg name="veh" doc="Name of vehicle. ex: megaman"/>


	<arg name="joystick" default="true" doc="true for listening to the joystick. False for listening to the coordination."/>
	<arg name="config" default="baseline" doc="Specify a config."/>
        <arg name="param_file_name" default="default" doc="Specify a param file. ex:megaman"/>

	<group ns="$(arg veh)">


	    	<node pkg="$(arg pkg_name)" type="$(arg src_name)"  name="$(arg node_name)" output="screen" clear_params="true" required="true">
			<rosparam command="load" file="$(find duckietown)/config/$(arg config)/led_interpreter/LED_protocol.yaml"/>

        		<!-- run with joystick -->
		    	<remap from="~change_color_pattern" to="led_joy_mapper_node/change_color_pattern" if="$(arg joystick)"/>
		    	<!-- run with coordination node -->
		    	<remap from="~change_color_pattern" to="explicit_coordination/change_color_pattern" unless="$(arg joystick)"/>
		</node>
	</group>
</launch>
